app/services/ScanningService.scala

```diff
@@ -29,26 +29,33 @@ import util.logging.LoggingDetails
 
 import scala.concurrent.Future
 
-trait ScanningService extends FileChecker
+trait ScanningService {
+  def scan(
+    location: S3ObjectLocation,
+    objectContent: ObjectContent,
+    objectMetadata: InboundObjectMetadata): Future[FileCheckingResultWithChecksum]
+}
 
 class ClamAvScanningService @Inject()(clamClientFactory: ClamAntiVirusFactory, metrics: Metrics)
     extends ScanningService {
 
   override def scan(
     location: S3ObjectLocation,
     fileContent: ObjectContent,
-    metadata: InboundObjectMetadata): Future[FileCheckingResult] = {
+    metadata: InboundObjectMetadata): Future[FileCheckingResultWithChecksum] = {
     implicit val ld = LoggingDetails.fromS3ObjectLocation(location)
 
     val antivirusClient       = clamClientFactory.getClient()
     val startTimeMilliseconds = System.currentTimeMillis()
 
+    val inputStream = new MessageDigestComputingInputStream(fileContent.inputStream, "SHA-256")
```

Comment:
Is there a way this can be injected into the concrete class, rather than new-ed up within the class?